_posts/2019-09-20-lambda-to-openfaas.md

```diff
@@ -0,0 +1,330 @@
+
+In this post, We'll walk through the steps taken to migrate existing AWS Lambda functions to OpenFaaS compatible functions in order to allow easier local testing, familiar development process, and the freedom to use any number of other cloud providers without any code changes. We won't be moving all of the infrastructure, only the functions. We'll keep the DynamoDB table as well as the IAM roles/permissions to access that data.
+
+Without further ado, let's see what it takes to upgrade your AWS Lambda functions to OpenFaaS and Serverless 2.0 using OpenFaaS!
+
+## Prerequisites
+
+- Existing AWS Lambda functions (optional)
+  - For the tutorial, we'll be using the Lambda function in this [Github repository](https://github.com/burtonr/lambda-openfaas-blog)
+- OpenFaaS 
+  - Check the documentation for [how to deploy to Kubernetes](https://docs.openfaas.com/deployment/kubernetes/)
+- OpenFaaS CLI
+  - Check the documentation for [installation instructions](https://docs.openfaas.com/cli/install/)
+- IDE, or other text editor
+  - I recommend [Visual Studio Code](https://code.visualstudio.com/)
+
+## Getting Started
+
+To begin, let's look at the "hello world" sample function that gets created when you get started with Lambda. When you follow the [Getting Started Guide](https://docs.aws.amazon.com/lambda/latest/dg/getting-started-create-function.html) in the AWS documentation, a basic sample function is generated for you in Node.js 10.x as shown below:
+
+```js
+exports.handler = async (event) => {
+    // TODO implement
+    const response = {
+        statusCode: 200,
+        body: JSON.stringify('Hello from Lambda!'),
+    };
+    return response;
+};
+```
+
+Similarly, when you create a new function with the OpenFaaS CLI and the `node10-express` template, sample code is provided as shown here:
+```js
+"use strict"
+
+module.exports = (event, context) => {
+    let err;
+    const result =             {
+        status: "You said: " + JSON.stringify(event.body)
+    };
+    context
+        .status(200)
+        .succeed(result);
+}
+```
+
+At a quick glance, the two function samples are very similar. The main difference is in the invocation parameters. OpenFaaS passes in the `context` object which is used to provide additional context to the result. The context includes the properties `status`, `succeed`, `error`, and others that make discovering functionality and managing the response of the function easy without having to dig through pages of documentation.
+
+As you can see, migrating the sample function from Lambda to OpenFaaS would be as easy as adding the `context` object to the invocation, and moving the values from the inline `response` object to the OpenFaaS included `context` object properties.
+
```

Comment:
`You can see the documentation and other examples for the node10-express template here.` <- add this and a link